### PR TITLE
Remove duplicated suggestions in awesome bar

### PR DESCRIPTION
**Pull request by HollowMan6**
We now only remove duplicated suggestions based on URL, maybe later we can decide to remove those entries that only differ in 'http'/'https', or end with '/' or not

Before:
image

After:
image

### CODE REVIEW

File: app/src/common/shared/com/igalia/wolvic/search/suggestions/SuggestionsProvider.java [hunk 1]
```diff
@@ -180,11 +184,60 @@ private CompletableFuture<List<SuggestionItem>> getSearchEngineSuggestions(@NonN
...
+        CompletableFuture<List<SuggestionItem>> future = new CompletableFuture<>();
+
+        Map<String, List<Integer>> category = new HashMap<>();
+        Map<String, Integer> best = new HashMap<>();
+
+        // Filter out duplicate items based on the URL
+        for (int i = 0; i < items.size(); i++) {
+            SuggestionItem thisItem = items.get(i);
+            // Normalize URLs to remove 'http'/'https' and trailing slashes
+            String normalizedUrl = thisItem.url.replaceAll("^https?://", "").replaceAll("/$", "");
```
Comment (javifernandez) [inline app/src/common/shared/com/igalia/wolvic/search/suggestions/SuggestionsProvider.java:197]: Perhaps we could get advantage of the utility functions we have in the UrlUtil class. Additionally, we could try to create a java.net.URL instance from the string and uses it's API to categorize the SutestionItem instance.

Comment (HollowMan6) [reply app/src/common/shared/com/igalia/wolvic/search/suggestions/SuggestionsProvider.java]: Added some logic to handle this in UrlUtils

**Review by svillar — CHANGES_REQUESTED** [review]
I understand the rationale of the change but I am not sure it can be successfully resolved by filtering. For example imagine that you have a bookmark with a trailing slash and also a history entry without the slash. Then the history entry will be shown, but not the bookmark. That for me is a mistake because bookmarks must be already shown IMO.

The other comment I have is that comparing strings is normally slow, so I'd use a hash of the URL to be inserted in the Map instead of the full URL.

Last but not least, I don't think we can freely assume that we can remove insecure schemes (like http). For example I have some personal services at home that can be accessed either by http or https as I don't really care too much since they are internal services. I'd presume that is the same in corporate environments.

**Comment by HollowMan6** [comment]
> I understand the rationale of the change but I am not sure it can be successfully resolved by filtering. For example imagine that you have a bookmark with a trailing slash and also a history entry without the slash. Then the history entry will be shown, but not the bookmark. That for me is a mistake because bookmarks must be always shown IMO.
> 
> The other comment I have is that comparing strings is normally slow, so I'd use a hash of the URL to be inserted in the Map instead of the full URL.
> 
> Last but not least, I don't think we can freely assume that we can remove insecure schemes (like http). For example I have some personal services at home that can be accessed either by http or https as I don't really care too much since they are internal services. I'd presume that is the same in corporate environments.

Okay, anyway we can revert back to https://github.com/Igalia/wolvic/commit/5929cde663415e3ddf9cf48c05fa141bc4506034 to make it less aggressive, and modify that so bookmarks must be always shown.

**Comment by HollowMan6** [comment]
> > I understand the rationale of the change but I am not sure it can be successfully resolved by filtering. For example imagine that you have a bookmark with a trailing slash and also a history entry without the slash. Then the history entry will be shown, but not the bookmark. That for me is a mistake because bookmarks must be always shown IMO.
> > The other comment I have is that comparing strings is normally slow, so I'd use a hash of the URL to be inserted in the Map instead of the full URL.
> > Last but not least, I don't think we can freely assume that we can remove insecure schemes (like http). For example I have some personal services at home that can be accessed either by http or https as I don't really care too much since they are internal services. I'd presume that is the same in corporate environments.
> 
> Okay, anyway we can revert back to [5929cde](https://github.com/Igalia/wolvic/commit/5929cde663415e3ddf9cf48c05fa141bc4506034) to make it less aggressive, and modify that so bookmarks must be always shown.

Reverted. We use hashSet so it's fast to check if a URL already exists. Also we sort before we remove duplication so that only those duplicated results that appear at the bottom of the list will get removed

**Review by svillar — APPROVED** [review]
This more conservative version looks more appealing to me. Thanks!

**Pull request merged**